crates/sui-archival/src/blob.rs

```diff
@@ -0,0 +1,549 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Context, Result};
+use byteorder::{BigEndian, ByteOrder, ReadBytesExt};
+use crc::{Crc, CRC_32_ISCSI};
+use integer_encoding::VarInt;
+use integer_encoding::*;
+use num_enum::IntoPrimitive;
+use num_enum::TryFromPrimitive;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::fs::{read_dir, File, OpenOptions};
+use std::io::{BufReader, BufWriter, Read, Seek, SeekFrom, Write};
+use std::path::PathBuf;
+use std::{fs, mem};
+use tracing::info;
+use tracing::log::error;
+
+/// The following describes the format of a blob file, which is created in the `transactions/`,
```

Comment:
Ok, so I think to summarize our discussion from today:
- Lets align to epoch boundaries snapshots for sure, but also lets add epoch to the path for snapshots and not have blobs spanning across epochs.
- Let's merge checkpoints, transactions, effects, and events into one hierachy. So within blobs we would have (1) checkpoint (2) followed by a sequence of triplets of type (transaction, effects, events). A process that reads them in this order can authenticate them step by step.
- Objects should have two folders: references and content, and the snapshops should be by epoch. Each blob references should be a shard / partition aligned with the authenticators in checkpoints, and the references should match the authenticators. Then the object content blob for the epoch should contain the contents of these objects that can be authenticated by reference.
- Lets drop the CRCs, both the checkpoint/transaction/effects/event blobs can be authenticated by signatures and hashes, and so can the references / objects.

Reply:
Thanks for summarizing it so well. I think we are aligned on all things. I will drop CRCs like we discussed.